### DISCUSSION TITLE
Flag image

### DISCUSSION THREAD

**Discussion by ashish-gaikwaad** [Daniel-Ioannou/flutter_country_picker#68] [Q&A] [answered]
How can i get the Flag Image

**Comment by Daniel-Ioannou** [comment, marked as answer]
@ashishgkwd534 
The package use emoji to show the flags (not images) you can get the flags emoji with `Utils.countryCodeToEmoji(country.countryCode;`

Example: 
```Dart
import 'package:country_picker/src/utils.dart';

Text(
     Utils.countryCodeToEmoji(country.countryCode),
     style: const TextStyle(fontSize: 25),
 );
```

Note that Utils is not expose to the package so you have to import it: `import 'package:country_picker/src/utils.dart';`

**Reply by abhinandval** [reply]
I wrote a simple extension to make things quick for me. 

```
import 'package:country_picker/src/utils.dart' as flag_util;
...
extension on Country {
  String get flagEmoji => flag_util.Utils.countryCodeToEmoji(countryCode);
}
```

Will be helpful if you can add this to the Country Model as a static function

check this PR #76

**Reply by Daniel-Ioannou** [reply]
Thank @abhinandval for your contribution. PR #76 released in [v2.0.13](https://pub.dev/packages/country_picker/versions/2.0.13).

**Comment by rennylangat** [comment]
> @ashishgkwd534 The package use emoji to show the flags (not images) you can get the flags emoji with `Utils.countryCodeToEmoji(country.countryCode;`
> 
> Example:
> 
> ```dart
> import 'package:country_picker/src/utils.dart';
> 
> Text(
>      Utils.countryCodeToEmoji(country.countryCode),
>      style: const TextStyle(fontSize: 25),
>  );
> ```
> 
> Note that Utils is not expose to the package so you have to import it: `import 'package:country_picker/src/utils.dart';`

Amazing thanks

**Comment by Nishtha-Iwebservices** [comment]
@Daniel-Ioannou  how to access country.countrycode out of showCountryPicker?

**Reply by Daniel-Ioannou** [reply]
Not sure what you mean by "access country.country code".

Country picker has callback function(`onSelect`) that provide an instance of  `Country` model  with the country selected by the user.

```Dart
showCountryPicker(
  context: context,
  onSelect: (Country country) {
    print('Select country: ${country.countrycode}');
  },
);
```

If you mean to get an instance of Country model by country code or name without the user choosing a country from CountryPicker, then you can use `CountryParser`. Check this discussion #67 or the [documentation](https://pub.dev/documentation/country_picker/latest/country_picker/CountryParser-class.html).

**Comment by rennylangat** [comment]
@Nishtha-Iwebservices All you have to do is declare a late variable String to hold your country code of initialize it with your default country. Inside the showCountryPicker, setState of the declared variable to the selected country

```
showCountryPicker( 
          context: context,
          onSelect: ((Country country) {
                         setState(() {
                                      countryCode = country.countryCode;
                        });
                }));
